src/main/java/bio/terra/cloudfunctions/common/GoogleCloudStorageEventHarness.java

```diff
@@ -0,0 +1,69 @@
+package bio.terra.cloudfunctions.common;
+
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.gson.internal.LinkedTreeMap;
+import java.util.logging.Logger;
+
+/**
+ * This class is an abstract representation of a Function triggered by a Google Cloud Storage Event.
+ * The function casts the Google Cloud Storage Event into a Map of key-value pairs.
+ *
+ * <p>This class contains only framework-specific logic to receive and parse Google Storage Object
+ * Event messages.
+ *
+ * <p>Sub-classes of this class can be integrated with a DI framework to deploy Cloud Functions like
+ * a Service.
+ */
+public abstract class GoogleCloudStorageEventHarness
+    implements BackgroundFunction<LinkedTreeMap<?, ?>> {
+  private static final Logger logger =
+      Logger.getLogger(GoogleCloudStorageEventHarness.class.getName());
+
+  private Context context;
+  private LinkedTreeMap<?, ?> event;
```

Comment:
What do we gain by using LinkedTreeMap to store the event data?  I think this class should implement `RawBackgroundFunction` and store the event as a string and then deserialize from that string in `getEvent()`

Reply:
See my comments yesterday. `LinkTreeMap` has been replaced by `Map` interface.